rotkehlchen/chain/ethereum/modules/curve/balances.py

```diff
@@ -0,0 +1,89 @@
+from collections import defaultdict
+from typing import TYPE_CHECKING, Callable
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.accounting.structures.evm_event import EvmProduct
+from rotkehlchen.accounting.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.assets.asset import EvmToken
+from rotkehlchen.chain.ethereum.interfaces.balances import ProtocolWithBalance
+from rotkehlchen.chain.evm.tokens import get_chunk_size_call_order
+from rotkehlchen.chain.evm.types import WeightedNode
+from rotkehlchen.db.dbhandler import DBHandler
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.types import ChecksumEvmAddress
+from rotkehlchen.utils.misc import get_chunks
+
+from .constants import CPT_CURVE
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+
+
+def query_gauges_balances(
+        user_address: ChecksumEvmAddress,
+        gauges_to_token: dict[ChecksumEvmAddress, EvmToken],
+        call_order: list[WeightedNode],
```

Comment:
This is executed inside a for loop for each address tracked so to avoid executing that logic I calculate it once and reuse it for each address tracked